DOC 4059

jamesnocentini opened the pull request:
https://issues.couchbase.com/browse/DOC-4059

jamesnocentini committed:
DOC-4059

https://issues.couchbase.com/browse/DOC-4059

rajagp commented on modules/ROOT/pages/swift.adoc:
```diff
@@ -236,6 +238,31 @@ The following example enables `verbose` logging for the `replicator` and `query`
+File based logging::
+File based logging is supported in two formats:
+
+* a plaintext mode
```
List this after binary mode since thats default

jamesnocentini replied:
https://github.com/couchbase/docs-couchbase-lite/commit/aea18dc924223dd975e2b91ea7b422ad3388f328

rajagp commented on modules/ROOT/pages/swift.adoc:
```diff
@@ -236,6 +238,31 @@ The following example enables `verbose` logging for the `replicator` and `query`
++
+It is recommended that you use binary log format.
+However, in order to view the binary encoded log files, you will need a decoder.
+To decode the binary logs, there is a binary log decoder called the *cbl-log* which is a tool downloadable from https://www.couchbase.com/downloads[here].
```
You should point this to the docs page which provides the install instructions for cb-log tool . We don't want to take users out of documentation site and no reason for users to have to search through downloads page to just get the link to documentation site.  As FYI - the downloads page  simply has link to the documentation site - nothing more

jamesnocentini replied:
https://github.com/couchbase/docs-couchbase-lite/commit/1d30b75458ce604f89054265cb07083c70c2ab3a - FYI the cbl-log section is empty atm (will add content as part of another ticket)

rajagp commented on modules/ROOT/pages/swift.adoc:
```diff
@@ -236,6 +238,31 @@ The following example enables `verbose` logging for the `replicator` and `query`
+To decode the binary logs, there is a binary log decoder called the *cbl-log* which is a tool downloadable from https://www.couchbase.com/downloads[here].
++
+Custom logging::
+Provides apps the ability to use other logging frameworks to handle Couchbase Lite log messages.
```
Please provide an example for each platform. Apps will have to implement this interface
```
interface Logger {
 var level: LogLevel

 func log(
    level: LogLevel,  
    domain: LogDomain,
    message: String)
}
```

jamesnocentini replied:
https://github.com/couchbase/docs-couchbase-lite/commit/ac72abaf4f1dbabf202b3e54dd421b353c609ad9 - just a placeholder for now. I have a few questions which we can work through on the docs sync up call.

rajagp commented on modules/ROOT/pages/swift.adoc:
```diff
@@ -236,6 +238,31 @@ The following example enables `verbose` logging for the `replicator` and `query`
+To decode the binary logs, there is a binary log decoder called the *cbl-log* which is a tool downloadable from https://www.couchbase.com/downloads[here].
++
+Custom logging::
+Provides apps the ability to use other logging frameworks to handle Couchbase Lite log messages.
```
```suggestion
Provides apps the ability to register callback function in order to receive Couchbase Lite log messages. The messages can be logged using any external logging framework.
```

bmeike committed:
Import working java

jamesnocentini committed:
change ordering

jamesnocentini committed:
Point to docs section w/ install instructions

rajagp committed:
Update modules/ROOT/pages/swift.adoc

Co-Authored-By: jamiltz <james.nocentini@gmail.com>

jamesnocentini committed:
fix syntax

jamesnocentini committed:
placeholder for custom logging code example

jamesnocentini committed:
DOC-4798 / https://issues.couchbase.com/browse/DOC-4798

jamesnocentini committed:
update per review

jamesnocentini committed:
update per review, add Swift/Java code example

jamesnocentini committed:
update per review

jamesnocentini committed:
update per review

jamesnocentini committed:
move logging docs to partial

jamesnocentini committed:
add logging content to c#, java and objc guides

Commit:
Add File logging and Custom logging example in .Net

jamesnocentini commented on modules/ROOT/examples/csharp/Program.cs:
```diff
@@ -946,4 +962,27 @@ public async Task Send(Message message)
+        [NotNull]
+        private readonly List<string> _lines = new List<string>();
+
+        public IReadOnlyList<string> Lines => _lines;
```
@Sandychuang8 would it be possible to remove references to `lines` to match the Swift version https://github.com/couchbase/docs-couchbase-lite/pull/71/files#diff-dc21fc8be6ee8786054080415959e410R728 ? I believe `lines` is used in unit tests.

Commit:
Remove line reference

bmeike committed:
DOC-5129 / https://issues.couchbase.com/browse/DOC-5129

Conflicts:
	modules/ROOT/examples/java/app/src/main/java/com/couchbase/code_snippets/MainActivity.java

bmeike committed:
DOC-5129 / https://issues.couchbase.com/browse/DOC-5129

jamesnocentini committed:
Revert "Import working java"

This reverts commit 301b2b649aed9db53aeb2b5238f11176b653d64c.

Conflicts:
	modules/ROOT/examples/java/app/src/main/java/com/couchbase/code_snippets/MainActivity.java

jamesnocentini committed:
revert changes

jayahariv commented on modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift:
```diff
@@ -57,6 +57,20 @@ class SampleCodeTest {
+    func dontTestFileLogging() throws {
+        // tag::file-logging[]
+        let tempFolder = NSTemporaryDirectory().appending("cbllog")
+        Database.log.file.config = LogFileConfiguration.init(directory: tempFolder)
```
not much difference, but more easy = `LogFileConfiguration(directory: tempFolder)`

jamesnocentini replied:
Yes. Updated in https://github.com/couchbase/docs-couchbase-lite/commit/ae30b84c9253d44e5ed4f11101eec45813c5b6e9
